presto-hive/src/main/java/com/facebook/presto/hive/CachingDirectoryLister.java

```diff
@@ -55,11 +56,13 @@ public CachingDirectoryLister(@ForCachingDirectoryLister DirectoryLister delegat
                 hiveClientConfig.getFileStatusCacheExpireAfterWrite(),
                 hiveClientConfig.getFileStatusCacheMaxSize(),
                 hiveClientConfig.getFileStatusCacheTables().stream()
+                        .filter(table -> !table.equals("*"))
                         .map(CachingDirectoryLister::parseTableName)
-                        .collect(Collectors.toSet()));
+                        .collect(Collectors.toSet()),
+                hiveClientConfig.getFileStatusCacheTables().contains("*"));
     }
 
-    public CachingDirectoryLister(DirectoryLister delegate, Duration expireAfterWrite, long maxSize, Set<SchemaTableName> tables)
+    public CachingDirectoryLister(DirectoryLister delegate, Duration expireAfterWrite, long maxSize, Set<SchemaTableName> tables, boolean cacheAllTables)
```

Comment:
Encapsulate `Set<SchemaTableName> tables` and `boolean cacheAllTables` into a private inner class to prevent cases where `tables` is non-empty and `cacheAllTables = true`